.buildkite/scripts/signAndPublishPackage.sh

```diff
@@ -8,10 +8,21 @@ source .buildkite/scripts/install_deps.sh
 source .buildkite/scripts/tooling.sh
 
 cleanup() {
+    local error_code=$?
+
+    if [ $error_code != 0 ] ; then
+        # if variable is defined, run the logout
+        if [ -n "${GOOGLE_APPLICATION_CREDENTIALS+x}" ]; then
+             google_cloud_logout_active_account
+        fi
+    fi
+
```

Comment:
Perhaps it's better to move the `cleanup` func into `tooling.sh` script to avoid duplicating the code in the `signAndPublishPackage.sh` and `integration_tests.sh`. @mrodm WDYT?

Reply:
I wanted to avoid that in this PR at least, I added a comment about this in this other thread:
https://github.com/elastic/elastic-package/pull/1487#discussion_r1347228719